quickstart-template/register-basename.js

```diff
@@ -0,0 +1,165 @@
+            type: "string",
+          },
+          {
+            internalType: "address",
+            name: "owner",
+            type: "address",
+          },
+          {
+            internalType: "uint256",
+            name: "duration",
+            type: "uint256",
+          },
+          {
+            internalType: "address",
+            name: "resolver",
+            type: "address",
+          },
+          {
+            internalType: "bytes[]",
+            name: "data",
+            type: "bytes[]",
+          },
+          {
+            internalType: "bool",
+            name: "reverseRecord",
+            type: "bool",
+          },
+        ],
+        internalType: "struct RegistrarController.RegisterRequest",
+        name: "request",
+        type: "tuple",
+      },
+    ],
+    name: "register",
+    outputs: [],
+    stateMutability: "payable",
+    type: "function",
+  },
+];
+
+// Basenames Registrar Controller Contract Address.
+const BaseNamesRegistrarControllerAddress = "0x4cCb0BB02FCABA27e82a56646E81d8c5bC4119a5";
+
+// Create register contract method arguments.
+function createRegisterContractMethodArgs(baseName, addressId) {
+  const addressData = encodeFunctionData({
+    abi: l2ResolverABI,
+    functionName: "setAddr",
+    args: [namehash(baseName), addressId],
+  });
+  const nameData = encodeFunctionData({
+    abi: l2ResolverABI,
+    functionName: "setName",
+    args: [namehash(baseName), baseName],
+  });
+
+  const registerArgs = {
+    request: [
+      baseName.replace(/\.base\.eth$/, ""),
+      addressId,
+      "31557600",
+      BaseNamesRegistrarControllerAddress,
+      [addressData, nameData],
+      true,
+    ],
+  };
+  console.log(`Register contract method arguments constructed: `, registerArgs);
+
+  return registerArgs;
+}
+
+// Register a Basename for the given Wallet.
+async function registerBaseName(wallet, registerArgs) {
+  try {
+    const contractInvocation = await wallet.invokeContract({
+      contractAddress: BaseNamesRegistrarControllerAddress,
+      method: "register",
+      abi: registrarABI,
+      args: registerArgs,
+      amount: 0.002,
+      assetId: Coinbase.assets.Eth,
+    });
+
+    await contractInvocation.wait();
+
+    console.log(`Successfully registered Basename ${registerArgs[0]} for wallet: `, wallet);
+  } catch (error) {
+    console.error(`Error registering a Basename for ${wallet}: `, error);
+  }
+}
+
+// Fetch a funded Wallet and load its Seed.
+async function fetchWalletAndLoadSeed(walletId, seedFilePath) {
+  try {
+    const wallet = await Wallet.fetch(walletId);
+    await wallet.loadSeed(seedFilePath);
+
+    console.log(`Successfully loaded funded wallet: `, wallet);
+    return wallet;
+  } catch (error) {
+    console.error(
+      `Error loading funded wallet ${walletId} from seed file ${seedFilePath}: `,
+      error,
+    );
+  }
+}
+
+async () => {
+  try {
+    const { BASE_NAME, WALLET_ID, SEED_FILE_PATH } = process.env;
```

Comment:
add a comment that it should be set in ?

Reply:
The Solution guide will have a section explaining how to set these.